Main.py

```diff
@@ -60,31 +66,59 @@ def main():
     
     while True:
         keys = PybulletAPI.getKeyboardEvents()
-        if ord('q') in keys and keys[ord('q')] & PybulletAPI.KEY_WAS_TRIGGERED:
-            break
+        if ord('p') in keys and keys[ord('p')] == PybulletAPI.KEY_WAS_TRIGGERED:
+            paused = not paused
+
+        lobster_pos, lobster_orn = simulator.robot.get_position_and_orientation()
+
+        desired_location = Translation.vec3_rotate_vector_to_local(lobster_orn, desired_location)
+        if ord('q') in keys and keys[ord('q')] == p.KEY_IS_DOWN:
```

Comment:
Could create a small function for checking if a specific key is down? 
```suggestion
        if is_key_down('q'):
```
```
is_key_down(key: string, keys: any):
   return ord(key) in keys and keys[ord(key)] == p.KEY_IS_DOWN

Reply:
I realize that I could also use `return keys.get(ord('q') == p.KEY_IS_DOWN', but is it okay to postpone this, since there is already a future branch that has moved this to the HigherLevelController, so that would give some annoying conflicts.